jaxrs/resteasy-jaxrs/src/main/java/org/jboss/resteasy/specimpl/ResteasyUriBuilder.java

```diff
@@ -519,65 +552,66 @@ protected URI buildUriFromMap(Map<String, ? extends Object> paramMap, boolean fr
 
    private String buildString(Map<String, ? extends Object> paramMap, boolean fromEncodedMap, boolean isTemplate, boolean encodeSlash)
    {
-      for (Map.Entry<String, ? extends Object> entry : paramMap.entrySet())

```

Comment:
Maybe useless check since paramMap key and value will be tested in replaceParameter(...) and replaceQueryStringParameter(...) methods.